layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -184,6 +184,49 @@ Dedicated (locked) windows are left untouched."
   (interactive "p")
   (spacemacs/rotate-windows (* -1 count)))
 
+(defun spacemacs/move-buffer-to-window (windownum follow-focus-p)
+  "Moves a buffer to a window, using the spacemacs numbering. follow-focus-p
+   controls whether focus moves to new window (with buffer), or stays on
+   current"
+  (interactive)
+  (let ((b (current-buffer))
+        (w (selected-window)))
+    (save-selected-window
```

Comment:
Well, note that in any case it switches the window, because it follows the active buffer. So in the case that's currently being bound, it makes no difference. I did leave the boolean parameter, for future use (buffer transient state, plus maybe to create an option to control the behavior). If in the future we call without wanting to change window focus, then yes what I'm suggested will involve two window switches instead of none.

Personally, I don't like implementing `get-window-by-number`, because we can't immediately change `select-window-by-number`, so now we have pretty bad code duplication. But I will do it your way.

Note that I was trying to get my stuff merged into window-number.el at one point; I saw that the project was inactive and I asked the maintainer if I could take it over. He got back to me several months later and said "yes", by which point I was no longer interested. If you are willing, you could email him. If you become the maintainer for window-number.el then we can merge some changes and get the best of both worlds (no unnecessary switching, no code duplication).